agent/utils/ephemeral_ports.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+// From https://www.kernel.org/doc/html/latest//networking/ip-sysctl.html#ip-variables
+const (
+	EphemeralPortMin         = 32768
+	EphemeralPortMax         = 60999
+	maxPortSelectionAttempts = 100
+)
+
+// GenerateEphemeralPortNumbers generates a list of n unique port numbers in the 32768-60999 range. The resulting port
+// number list is guaranteed to not include any port number present in "reserved" parameter.
+func GenerateEphemeralPortNumbers(n int, reserved []uint16) ([]uint16, error) {
```

Comment:
In addition to avoiding the "reserved" ports, there can be other processes on the host instance utilizing ephemeral ports. Although the chance for port contention in the ephemeral range can be relatively low, but it's still possible. And when it happens, the task won't be able to start. Therefore, I think we should have some checks here to see if the selected port is available on the host.

One way to do that is to try create a simple server with `net.Listen("tcp", ":" + port)`, and it will throw error if port is already in use. There might be other simpler ways that I'm not aware of.

Reply:
Discussed offline and clarified that those ports will be isolated to task netns. And because appnet is the first to start in task netns, there shouldn't be any port collision.